temporian/core/evaluator.py

```diff
@@ -52,72 +52,76 @@ def evaluate(
-        if not isinstance(input_event_spec, event):
-            input_event_spec = read_csv_fn(
-                input_event_spec, input_event.sampling()
-            )
-        materialized_input_data[input_event] = input_event_spec
-
+    materialized_input_data = {
+        input_event: (
+            input_event_spec
+            if isinstance(input_event_spec, event)
+            else read_csv_fn(input_event_spec, input_event.sampling())
+        )
+        for input_event, input_event_spec in input_data.items()
+    }
     # evaluate schedule
     outputs = evaluate_schedule_fn(materialized_input_data, schedule)
 
     return {event: outputs[event] for event in events_to_compute}
 
 
-def get_operator_schedule(query: List[Feature]) -> List[base.Operator]:
-    # TODO: add depth calculation for parallelization
-    # TODO: Make "query" a Set
-
-    operators_to_compute = []  # TODO: implement as ordered set
-    visited_features = set()
-    pending_features = list(query.copy())  # TODO: implement as ordered set
-    while pending_features:
-        feature = next(iter(pending_features))
-        visited_features.add(feature)
-
-        if feature.creator() is None:
-            # is input feature
-            pending_features.remove(feature)
-            continue
-
-        # required input features to compute this feature
-        creator_input_features = {
-            input_feature
-            for input_event in feature.creator().inputs().values()
-            for input_feature in input_event.features()
-        }
-
-        # check if all required input features have already been visited
-        if creator_input_features.issubset(visited_features):
-            # feature can be computed - remove it from pending_features
-            pending_features.remove(feature)
-
-            # add operator at the end of operators_to_compute
-            if feature.creator() not in operators_to_compute:
-                operators_to_compute.append(feature.creator())
-
-            continue
-
-        # add required input features at the beginning of pending_features
-        pending_features = list(creator_input_features) + pending_features
-
-    return operators_to_compute
+def get_operator_schedule(query: Set[Feature]) -> Dict[base.Operator, int]:
+    """Calculates which operators need to be executed in which order to
+    compute a set of query features.
+
+    Args:
+        query: set of query features to be computed.
+
+    Returns:
+        Dict[base.Operator, int]: dictionary mapping operators to their respective
+        depths in the compute graph. Depth is measured from bottom to top, i.e.
+        0 depth corresponds to the output features (query). Operators with the
+        same depth can be computed in parallel.
+    """
+    # start features. Depth initialized as 0
+    feature_depth = [(feature, 0) for feature in query]
+
+    # get all features and depths required to compute the query. One feature can
```

Comment:
Super Nit; Here and in other place: It could help the reading if sentence comments (i.e. comments that ends with a ".") has a first letter upper-case.